ksp_plugin/pile_up.hpp

```diff
@@ -30,10 +36,41 @@ class PileUp final {
  public:
   explicit PileUp(std::list<not_null<Vessel*>>&& vessels);
 
+  void set_mass_and_intrinsic_force(
+      Mass const& mass,
+      Vector<Force, Barycentric> const& intrinsic_force);
+
   std::list<not_null<Vessel*>> const& vessels() const;
 
+  // Flows the history authoritatively as far as possible up to |t|, advances
+  // the histories of the vessels.  After this call, the histories of |*this|
+  // and of its vessels have a (possibly ahistorical) final point exactly at
+  // |t|.
+  void AdvanceTime(
+      Ephemeris<Barycentric>& ephemeris,
+      Instant const& t,
+      Ephemeris<Barycentric>::FixedStepParameters const& fixed_step_parameters,
+      Ephemeris<Barycentric>::AdaptiveStepParameters const&
+          adaptive_step_parameters);
+
  private:
   std::list<not_null<Vessel*>> vessels_;
+  Mass mass_;
+  Vector<Force, Barycentric> intrinsic_force_;
+  DiscreteTrajectory<Barycentric> psychohistory_;
+  // True if the last point of the |psychohistory_| should be flowed from;
+  // otherwise, the last point should be removed before flowing the trajectory
+  // (in that case, there is a penultimate point, and it is historical).
+  bool psychohistory_is_history_;
+  // The |PileUp| is seen as a (currently non-rotating) rigid body; the degrees
+  // of freedom of the vessels in the frame of that can be set, however their
+  // motion is not integrated; this is simply applied as an offset from the
+  // rigid body motion of the |PileUp|.
+  using RigidPileUp = Frame<serialization::Frame::PluginTag,
+                            serialization::Frame::RIGID_PILE_UP,
+                            /*frame_is_inertial=*/false>;
```

Comment:
This needs a rigorous description of the frame `RIGID_PILE_UP`.  I assume that its origin is at the centre of mass of the pile-up, but I have no idea about its orientation.